### ISSUE TITLE
Sparkster; Wall-clipping Glitch

### ISSUE THREAD

**Issue by Kaleidosium** [elzo-d/SnesJs#2]
For some reason when I was testing your emulator with Sparkster, I was able to wallclip, and then was teleported to somewhere else in the game, I wasn't able to get any screenshots of this happening, but it did happen in the first try, It seems like collision is broken somehow, not sure if this is an in game bug or a bug in your emulator. I may have to ask byuu about this.

**Comment by angelo-wf** [comment]
I played a little bit of Sparkster, and although I did not notice any wallclipping specifically, there were definitely some bugs, with broken tiles showing on screen and weird camera movement.

This is most likely a problem with my emulator and not a bug in the game. There are probably still quite a lot of bugs and inaccuracies with (H)DMA, interrupts and timing that are causing these games to fail.

**Comment by Kaleidosium** [comment]
Correct, is there anyway I can contact you directly?, I'd love to test more games, but GitHub Issues aren't really instant feedback.

**Comment by angelo-wf** [comment]
To be honest, I don't know if I really want to put much more work into this emulator, at least not in the near future. Although it is a nice proof of concept, Javascript isn't really the best language to write emulators in, at least not for something as complicated like the SNES.

I mostly started it because my Javascript NES emulator ended up working quite nicely, and I thought a SNES emulator would therefor also be a nice project to do. It, however, turned out to be a good bit more difficult, especially timing wise. I am also not really sure how to properly debug the issues that it still has.

**Comment by Kaleidosium** [comment]
True, you may want to try writing it in a language like Rust/Go, They're much nicer for Emulation Development.
